Add example of using <details> tag in Markdown kitchen sink

blaylockbk opened the pull request:
I hope you will consider adding this example of how to use the `<details>` tag to the Markdown kitchen sink. I found the details in https://github.com/just-the-docs/just-the-docs/issues/246 solved a problem I was having, and adding this example would be a handy reference for the page.

max06 commented:
Hello @blaylockbk and welcome to just-the-docs!

Thank you very much for your contribution!

It looks good to me - I'm just wondering if the added style is required for it to function?

blaylockbk commented:
The `markdown='block'` is required, as shown here: https://github.com/just-the-docs/just-the-docs/issues/246#issuecomment-643783307

But the extra style isn't required. I just added it as a visual separator of the `<details>` block from the rest of the content. Would you prefer it was removed?

max06 commented:
In my opinion, it should be the least amount of code required, so others don't get confused.

About that end block... I see no purpose of that one, I'm fine with removing it.

blaylockbk commented:
Got it. I removed the style, and changed the name from "dropdown" to "collapsed section" because that's what [GitHub calls it](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections).

max06 commented:
Thank you very much for your contribution, we highly appreciate it! 
🖖🏼
